nc/source/graphics/vulkan/Resources/ResourceManager.h

```diff
@@ -0,0 +1,165 @@
+#pragma once
```

Comment:
Some of these functions try to move const string&, which doesn't do anything. I would get rid of the moves and keep the parameters as const& because eventually you're calling at(), which takes a const&.